docs/self-managed/setup/deploy/amazon/amazon-eks/eksctl.md

```diff
@@ -291,148 +298,256 @@ For detailed examples, review the [documentation provided by AWS](https://docs.a
+
+   ```shell
+   aws ec2 authorize-security-group-ingress \
+     --group-id $GROUP_ID \
+     --protocol tcp \
+     --port 443 \
+     --cidr $CIDR  # Replace with the CIDR range of your EKS cluster, e.g., <EKS_CIDR_BLOCK>
+   ```
+
+   Ensure that the CIDR range is appropriate for your environment.
+
+4. **Retrieve the private subnets of the VPC:**
+
+   ```shell
+   export SUBNET_IDS=$(aws ec2 describe-subnets \
+     --filter Name=vpc-id,Values=$VPC_ID \
+     --query "Subnets[?Tags[?Key=='aws:cloudformation:logical-id']|[?contains(Value, 'Private')]].SubnetId" \
+     --output text | expand -t 1)
+   ```
+
+   The variable `SUBNET_IDS` now contains the output values of the private subnets (the value should look like this: `subnet-0123456789 subnet-1234567890`).
+
+5. **Create the OpenSearch domain:**
+
+   ```shell
+   export OPENSEARCH_DOMAIN_NAME=<your-domain-name>  # Replace with your desired domain name
+
+   aws opensearch create-domain --domain-name $OPENSEARCH_DOMAIN_NAME \
+     --elasticsearch-version 2.15 \
+     --node-to-node-encryption-options Enabled=true \
+     --encryption-at-rest-options Enabled=true \
+     --vpc-options SubnetIds=[${SUBNET_IDS}],SecurityGroupIds=[${GROUP_ID}]
+
+     # TODO: add password and user
+     # TODO: use opensearch, not elastic
+   ```
+
+6. **Wait for the OpenSearch domain to be active:**
+
+   ```shell
+   aws opensearch wait domain-active --domain-name $OPENSEARCH_DOMAIN_NAME
+   ```
+
+7. **Retrieve the endpoint of the OpenSearch domain:**
+
+   ```shell
+   export OPENSEARCH_ENDPOINT=$(aws opensearch describe-domains --domain-names $OPENSEARCH_DOMAIN_NAME --query "DomainStatusList[0].Endpoint" --output text)
+   ```
+
+   This endpoint will be used to connect to your OpenSearch domain.
+
+### Verifying Connectivity from Within the EKS Cluster
+
+To verify that the OpenSearch domain is accessible only from within your Amazon EKS cluster, follow these steps:
+
+1. **Deploy a temporary pod to test connectivity:**
+
+   Create a temporary pod using a lightweight image, like `curlimages/curl`, to test the connection to OpenSearch.
+
+   ```shell
+   kubectl run curl-opensearch --rm -i --tty --image curlimages/curl -- sh
+   ```
+
+2. **Within the pod, test connectivity to the OpenSearch domain:**
+
+   Once inside the pod, use the following command to test the connection to the OpenSearch endpoint:
+
+   ```shell
+   curl -XGET https://$OPENSEARCH_ENDPOINT
+   ```
+
+   If everything is set up correctly, you should receive a response from the OpenSearch service.
 
-Verify that the connection is successful.
+You have successfully set up an OpenSearch domain that is accessible only from within your Amazon EKS cluster. Be sure to customize security group rules and CIDR ranges according to your environment's requirements. For further details, refer to the [OpenSearch documentation](https://opensearch.org/docs/latest/index/).
```

Comment:
depending on how you rephrase the above suggestion, adjust the only or leave it depending on the outcome.